File: JavaOpenDataAccess/src/OpenDataAccess/Entity/IEntity.java
package OpenDataAccess.Entity;

import OpenDataAccess.LambdaInterface.IFunction2;

import java.util.List;

public interface IEntity {
    EntityType ToEntityType();

    IEntityData ToEntityData() throws  IllegalAccessException;

    IEntityData ToEntityData(List<String> fieldList)throws  IllegalAccessException;

    void InsertValidate(List<IFunction2<IValidate, IEntityData, String>> validateList);

    void UpdateValidate(List<IFunction2<IValidate, IEntityData, String>> validateList);

    void DeleteValidate(List<IFunction2<IValidate, IEntityData, String>> validateList);

    <T extends  IEntity> IFunction2<IValidate, IEntityData, String> InsertValidateUnique(String propertyName, String message);
    <T extends  IEntity> IFunction2<IValidate, IEntityData, String> UpdateValidateUnique(String propertyName, String message);
    <T extends  IEntity> IFunction2<IValidate, IEntityData, String> ValidateExists(String filter, String message, boolean blExists);
}

File: AbetAccount/AbetAccount_Api/OpenDataAccessCore/OpenDataAccessCore.Entity/IEntity.cs
using System;
using System.Collections.Generic;

namespace OpenDataAccessCore.Entity
{
    public interface IEntity
    {
        EntityType ToEntityType();
        IEntityData ToEntityData();
        IEntityData ToEntityData(List<string> fieldList);
        void InsertValidate(List<Func<IValidate, IEntityData, string>> validateList);
        void UpdateValidate(List<Func<IValidate, IEntityData, string>> validateList);
        void DeleteValidate(List<Func<IValidate, IEntityData, string>> validateList);
        Func<IValidate, IEntityData, string> InsertValidateUnique<T>(string propertyName, string message) where T : IEntity;
        Func<IValidate, IEntityData, string> UpdateValidateUnique<T>(string propertyName, string message) where T : IEntity;
        Func<IValidate, IEntityData, string> ValidateExists<T>(string filter, string message, bool blExists = true) where T : IEntity;
    }
}
